distributed/comm/addressing.py

```diff
@@ -9,16 +9,21 @@
 DEFAULT_SCHEME = config.get('default-scheme', 'tcp')
 
 
-def parse_address(addr):
+def parse_address(addr, strict=False):
     """
     Split address into its scheme and scheme-dependent location string.
 
     >>> parse_address('tcp://127.0.0.1')
     ('tcp', '127.0.0.1')
+
+    If strict is set to true the address must have a scheme.
     """
     if not isinstance(addr, six.string_types):
         raise TypeError("expected str, got %r" % addr.__class__.__name__)
     scheme, sep, loc = addr.rpartition('://')
+    if strict and not sep:
+        raise ValueError("Invalid url scheme. Must be of the form <protocol>://<host>:<port>. "
+                        "Got %s" % addr)
```

Comment:
> Any thoughts on how to test this

I had an idea about specifying a `listen-address` and an `advertise-address` and then checking if the process is actually connecting to the listen-address and the scheduler is actually connecting on the advertised-address. I think if I can somehow find out the process-id of the worker process this should not be hard. I am not sure about the scheduler part. Manually I just verify through the logs but not sure how to do it in this case.

> 1. For workers would people want to specify an address but not a port, asking for that to be randomly assigned by the OS? If so then does the current system accomplish this?

I believe currently, this is explicit. You need to specify the ports. I will post an update after checking.
> 2. Do we also want these keywords for the scheduler?

I don't think so. You can bind it to 0.0.0.0 and do the necessary redirection because the scheduler does not advertise its IP by itself.
> 3. Not all protocols have this structure. Consider for example the inproc:// protocol

I understand this. But the way the `cli/dask_worker.py` is written it expects a port? I can make the change but I am not clear about how to test this (even manually). I mean start the service on `inproc` address (is this supported?) but the scheduler and worker process are anyway on different machines and even if not there would be more workers that may need to communicate with this worker that could be on a different machine?